trie/trie.go

```diff
@@ -317,7 +344,12 @@ func (t *Trie) insert(n node, prefix, key []byte, value node) (bool, node, error
 		if matchlen == 0 {
 			return true, branch, nil
 		}
-		// Otherwise, replace it with a short node leading up to the branch.
+		// New branch node is created as a child of the original short node.
+		// Track the newly inserted node in the tracer. The node identifier
+		// passed is the path from the root node.
+		t.tracer.onInsert(append(prefix, key[:matchlen]...))
```

Comment:
how does this not `panic` if no tracer is used?

Reply:
It's in `tracer`. If the tracer is nil, all the operations are ignored silently.

Reply:
I thought that this might be costly -- even if the tracer is `nil`, there might be a function call + allocation for the data passed. However, I didn't notice anything special when running some benchmarks.